File: problems/C++/01. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(std::vector<int>& nums, int target) {
        unordered_map<int, int> map;
        
        for (int i = 0; i < nums.size(); ++i) {
            int diff = target - nums[i];
            if (map.count(diff)) {
                return {map[diff], i};
            }
            map[nums[i]] = i;
        }
        return {};
    }
};

File: problems/Rust/01. Two Sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();
        
        for (i, n) in nums.into_iter().enumerate() {
            let diff = target - n;
            if let Some(&idx) = map.get(&diff) {
                return vec![idx, i as i32];
            }
            map.insert(n, i as i32);
        }
        vec![]
    }
}